src/item/component/RenderOffsetsComponent.php

```diff
@@ -3,40 +3,46 @@
 
 namespace customiesdevs\customies\item\component;
 
-final class RenderOffsetsComponent implements ItemComponent {
 
-	private int $textureWidth;
-	private int $textureHeight;
-	private bool $handEquipped;
+final class RenderOffsetsComponent implements ItemComponent { //thanks to https://github.com/MedicalJewel105/bedrock-render-offsets-generator/blob/main/main.py
 
-	public function __construct(int $textureWidth, int $textureHeight, bool $handEquipped = false) {
-		$this->textureWidth = $textureWidth;
-		$this->textureHeight = $textureHeight;
-		$this->handEquipped = $handEquipped;
+	private int $textureSize;
+
+	public function __construct(int $textureSize) {
+		$this->textureSize = $textureSize;
 	}
 
 	public function getName(): string {
 		return "minecraft:render_offsets";
 	}
 
 	public function getValue(): array {
-		$horizontal = ($this->handEquipped ? 0.075 : 0.1) / ($this->textureWidth / 16);
-		$vertical = ($this->handEquipped ? 0.125 : 0.1) / ($this->textureHeight / 16);
-		$perspectives = [
-			"first_person" => [
-				"scale" => [$horizontal, $vertical, $horizontal],
+	        $textureSize = $this->textureSize;
+	        $mainHand_fp = round(0.039 * 16 / $textureSize, 8);
+	        $offhand_fp = round(0.065 * 16 / $textureSize, 8);
+	        $mainHand_tp = $offhand_tp = round(0.0965 * 16 / $textureSize, 8);
```

Comment:
Could you please not use snake case? something like `$mainHandFirst` and `$mainHandThird` would work

Reply:
That's some GPT magic, thx